lib/util/dump-collection.ts

```diff
@@ -0,0 +1,334 @@
+      this.options
+    );
+
+    do {
+
+      for (const hit of results.hits) {
+        await this.onResult({
+          _id: hit._id,
+          _source: hit._source
+        });
+      }
+    } while ((results = await this.scroll(results.scrollId)));
+
+    await this.tearDown();
+
+    this.writeStream.end();
+
+    return waitWrite;
+  }
+}
+
+class JSONLDumper extends AbstractDumper {
+  protected ndjsonStream = ndjson.stringify();
+
+  async setup () {
+    this.ndjsonStream.on('data', (line: string) => {
+      this.writeStream.write(line);
+    });
+  }
+
+  async writeHeader () {
+    await this.writeLine({
+      collection: this.collection,
+      index: this.index,
+      type: 'collection',
+    });
+  }
+
+  writeLine (content: any): Promise<void> {
+    return new Promise(resolve => {
+      if (this.ndjsonStream.write(content)) {
+        resolve();
+      }
+      else {
+        this.ndjsonStream.once('drain', resolve);
+      }
+    });
+  }
+
+  onResult (document: {_id: string, _source: any}): Promise<void> {
+    return this.writeLine({
+      _id: document._id,
+      body: document._source,
+    });
+  }
+
+  protected get fileExtension () {
+    return 'jsonl';
+  }
+}
+
+class CSVDumper extends AbstractDumper {
+  constructor (
+    index: string,
+    collection: string,
+    query: any = {},
+    writeStream: stream.Writable,
+    options: JSONObject,
+    protected fields: string[],
+  ) {
+    super(index, collection, query, writeStream, options);
+  }
+
+  protected get fileExtension (): string {
+    return 'csv';
+  }
+  async setup () {
+    if (! this.fields.length) {
+      // If no field has been selected, then all fields are selected.
+      const mappings = await global.kuzzle.ask(
+        'core:storage:public:mappings:get',
+        this.index,
+        this.collection
+      );
+      if (! mappings.properties) {
+        return;
+      }
+      this.fields = Object.keys(flattenObject(extractMappingFields(mappings.properties)));
+    }
+    else if (this.fields.includes('_id')) {
+      // Delete '_id' from the selected fields, since IDs are
+      // _always_ exported.
+      this.fields.splice(this.fields.indexOf('_id'), 1);
+    }
+  }
+  writeHeader () {
+    return this.writeLine(['_id', ...this.fields].join(this.options.separator));
+  }
+
+  writeLine (content: any): Promise<void> {
+    return new Promise(resolve => {
+      if (this.writeStream.write(`${content}\n`)) {
+        resolve();
+      }
+      else {
+        this.writeStream.once('drain', resolve);
+      }
+    });
+  }
+
+  onResult (document: { _id: string; _source: any }): Promise<void> {
+    const values = [document._id, ...pickValues(document._source, this.fields)];
+    return this.writeLine(values.join(this.options.separator));
+  }
+}
+
+export function dumpCollectionDocument (index: string, collection: string, query: any = {}, format = 'jsonl', fields: string[] = [], options: JSONObject = {}): HttpStream {
```

Comment:
Nitpicking: this should be pluralized
```suggestion
export function dumpCollectionDocuments (index: string, collection: string, query: any = {}, format = 'jsonl', fields: string[] = [], options: JSONObject = {}): HttpStream {
```